content/src/debug.js

```diff
@@ -66,17 +62,55 @@ var debug = window.ide = {
     return false;
   },
   reportEvent: function(name, data) {
+    console.log('reportEvent', name, data);
     if (!targetWindow) {
       return;
     }
-    // Based on profiling data, we dispatch by hand instead of with
-    // jQuery.trigger.  (This speeds up fractals by 40%.)
-    if (name == 'enter') { debugEnter.apply(null, data); }
-    else if (name == 'exit') { debugExit.apply(null, data); }
-    else if (name == 'appear') { debugAppear.apply(null, data); }
-    else if (name == 'resolve') { debugResolve.apply(null, data); }
-    else if (name == 'init') { debugInit.apply(null, data); }
-    else if (name == 'error') {
+
+    console.log("reportEvent", name, data);
+
+    if(name === "appear"){
+      var debugId = data[1];
```

Comment:
Instead of unpacking data[1], data[2], etc, consider defining a function for appear, and calling it.

It would be something like this:

```js
function debugAppear(method, debugId, arg2, coordId, elem, eventArgs) {
  // everything in here
}
```

and then call it using "apply" to unpack the array

```js
if (name == 'appear') { debugAppear.apply(null, data); }
```